torchdata/dataloader2/dataloader2.py

```diff
@@ -67,6 +67,11 @@ def __next__(self) -> T_co:
                 if self.dataloader.reading_service is not None:
                     self.dataloader.reading_service.finalize_iteration()
                 raise
+            except Exception:
```

Comment:
1. Is this potentially too broad? Should I limit this to specific exceptions? I wonder if users would want to intentionally catch exceptions that are raised here (such that finalization calls aren't needed).
2. We can also call `DataLoader2.shutdown()` here instead.

Thoughts?

Reply:
I think calling dl2.shutdown should be good